lib/bloc/menubloc/menu_bloc.dart

```diff
@@ -0,0 +1,96 @@
+import 'package:bloc/bloc.dart';
+import 'package:cloud_firestore/cloud_firestore.dart';
+import 'package:nutrilab/bloc/menubloc/menu_event.dart';
+import 'package:nutrilab/bloc/menubloc/menu_state.dart';
+import 'package:nutrilab/models/menuitemmodel.dart';
+
+class MenuBloc extends Bloc<MenuEvent, MenuState> {
+  final String userId;
+
+  MenuBloc(this.userId) : super(MenuInitial()) {
+    on<LoadMenuItems>(_onLoadMenuItems);
+  }
+
+  Future<void> _onLoadMenuItems(
+      LoadMenuItems event, Emitter<MenuState> emit) async {
+    emit(MenuLoading());
+
+    try {
+      DocumentSnapshot userSnapshot =
+          await FirebaseFirestore.instance.collection('users').doc(userId).get();
+      Map<String, dynamic> userData =
+          userSnapshot.data() as Map<String, dynamic>;
+
+      List<String> likedItems = List<String>.from(userData['likedItems'] ?? []);
```

Comment:
Better to move this strings 'likedItems', 'cartItems', 'users' in constants, for now you can just move it above class MenuBloc

Reply:
Unable to do that as userId is being passed into constructor. Maybe I will pass likedItems and cartItems into this bloc so no need to deal with user

Reply:
I mean hardcoded strings `userData['**likedItems**']`